CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift

```diff
@@ -47,17 +52,73 @@ public class PeerToPeerRequestHandler {
             // Peer to Peer Apis //
             ////////////////////////////
             
-        case "peerToPeer_serverStart":
+        case "peerToPeer_messageEndpointListenerStart":
             let database: Database = args.get(name:"database")!
             let port: Int = args.get(name:"port")!
             let peerToPeerListener: ReplicatorTcpListener = ReplicatorTcpListener(databases: [database], port: UInt32(port))
             peerToPeerListener.start()
+            return peerToPeerListener
+
+        case "peerToPeer_getListenerPort":
+            let listener: URLEndpointListener = args.get(name:"listener")!
+            return listener.port
+    
+        case "peerToPeer_serverStart":
+            var peerToPeerListener: URLEndpointListener?
+            let database: Database = args.get(name:"database")!
+            let wsPort: Int = args.get(name: "port")!
+            let tls_disable: Bool? = args.get(name: "tls_disable")!
+            let config = URLEndpointListenerConfiguration.init(database: [database][0])
+            let tlsAuthType: String = args.get(name:"tls_auth_type")!
+            let tlsAuthenticator: Bool? = args.get(name: "tls_authenticator")!
+            config.port = UInt16(wsPort)
+            config.disableTLS = tls_disable!
+            
+            if (tlsAuthenticator != false) {
+                let rootCertData = try dataFromResource(name: "identity/client-ca", ofType: "der")
+                let rootCert = SecCertificateCreateWithData(kCFAllocatorDefault, rootCertData as CFData)!
+                let listenerAuth = ListenerCertificateAuthenticator.init(rootCerts: [rootCert])
+                try! TLSIdentity.deleteIdentity(withLabel: "CBL-Cert")
+                config.authenticator = listenerAuth
+                print("Setting Authenticator >>>>>>>>>>>>>>>")
+            }
+            
+            if tlsAuthType == "self_signed" {
+                let serverCertData = try! dataFromResource(name: "identity/certs", ofType: "p12")
+                try! TLSIdentity.deleteIdentity(withLabel: "CBL-Cert")
+                let identity = try! TLSIdentity.importIdentity(withData: serverCertData, password: "123", label: "CBL-Cert")
+                config.tlsIdentity = identity
+                print("Setting identity >>>>>>>>>>>>>>>")
+
+            } else if tlsAuthType == "self_signed_create" {
+                try! TLSIdentity.deleteIdentity(withLabel: "CBL-Cert")
+                let id = try! TLSIdentity.createIdentity(forServer: true , attributes: [certAttrCommonName: "CBL-Server"], expiration: nil, label: "CBL-Cert")
+                config.tlsIdentity = id
+                print("Setting CreateIdentity >>>>>>>>>>>>>>")
+            }
+
+             print(args.get(name: "basic_auth")!)
```

Comment:
I saw there are quite many print statements. Did ios testserver app implement its own logging? java/android has it. I remember I have investigated, just wanna to know